src/librustc_interface/passes.rs

```diff
@@ -532,6 +535,16 @@ fn configure_and_expand_inner<'a>(
     Ok((krate, resolver))
 }
 
+fn parser() -> &'static dyn Parser {
+    struct Parse;
+    impl Parser for Parse {
+        fn nt_to_tokenstream(&self, nt: &Nonterminal, sess: &ParseSess, span: Span) -> TokenStream {
```

Comment:
Alternatively we can just use a function pointer which requires relatively speaking much less code.

Reply:
Changed to this.